packages/commandkit/src/handlers/command-handler/functions/checkDuplicateCommands.ts

```diff
@@ -0,0 +1,34 @@
+import { CommandFileObject } from '../../../typings';
+import colors  from '../../../utils/colors'
+export function checkDuplicateCommands(commands: CommandFileObject[]){
+    let tempCommands = [...commands], duplicates = [], nonduplicates = [];
+    for(let i = 0; tempCommands.length != 0;){
+      let indexes = []
+       let duplicatePairs = tempCommands.filter((obj, index) => {
+        if(obj.data.name === tempCommands[i].data.name){
+          indexes.push(index)
+          return true
+        }
+        else return false
+       })
+       if(duplicatePairs.length > 1) duplicates.push(duplicatePairs)
+       else nonduplicates.push(duplicatePairs)
+       for(let j = indexes.length - 1; j > -1; j--){
```

Comment:
can we simplify this more?

Reply:
The logic now creates an obj that will have a key and hold the CommandFileObjects in an array then check if the length of that key is above one. This code does look a lot cleaner now.